### PR TITLE
Improve GetArgumentName in MonoOptions. Fixes bug-60904

### PR DESCRIPTION
**Pull request by EgorBo**
Fixes https://bugzilla.xamarin.com/show_bug.cgi?id=60904
GetArgumentName didn't properly handle double braces (escaped braces)

### CODE REVIEW

File: mcs/class/Mono.Options/Mono.Options/Options.cs [hunk 1]
```diff
@@ -1390,28 +1390,30 @@ static void Write (TextWriter o, ref int n, string s)
...
-				} while (start >= 0 && j != 0 ? description [j++ - 1] == '{' : false);
-				if (start == -1)
-					continue;
-				int end = description.IndexOf ("}", start);
-				if (end == -1)
-					continue;
-				return description.Substring (start + nameStart [i].Length, end - start - nameStart [i].Length);
+			var matches = Regex.Matches (description ?? "", @"(?<=(?<!\{)\{)[^{}]*(?=\}(?!\}))"); // ignore double braces 
+			string argName = "";
+			foreach (Match match in matches)
```
Comment (marek-safar) [inline mcs/class/Mono.Options/Mono.Options/Options.cs]: Keep the curly braces formatting consistent

File: mcs/class/Mono.Options/Mono.Options/Options.cs [hunk 2]
```diff
@@ -1390,28 +1390,30 @@ static void Write (TextWriter o, ref int n, string s)
...
+			{
+				var parts = match.Value.Split (':');
+				if (maxIndex == 1)
+				{
+					// for maxIndex=1 it can be {foo} or {0:foo}
+					argName = parts[parts.Length - 1];
+				}
+				// look for {i:foo} if maxIndex > 1
+				if (maxIndex > 1 && parts.Length == 2 && parts[0] == index.ToString ())
```
Comment (marek-safar) [inline mcs/class/Mono.Options/Mono.Options/Options.cs]: ToString should use InvariantCulture formatting

**Comment by luhenry** [comment]
@monojenkins build Linux AArch64

**Comment by luhenry** [comment]
@monojenkins build Linux AArch64 Interpreter

**Comment by luhenry** [comment]
@monojenkins build Linux x64

**Pull request merged**